TMS.Net07.HomeworkDateToDayOfWeeks/TMS.Net07.Homework2 - middle/Program.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TMS.Net07.Homework2___middle
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string input;
+            DateTime inputDate, maxRange = DateTime.Parse("31.12.2999");
+
+            Console.Title = "Dmitry Salata - TMS.Net07.Homework2 - level not slave, but not master (middle)";
+            Console.WriteLine("Please, enter the date in format DD.MM.yyyy or DD/MM/yyyy.");
+
+            input = Console.ReadLine();
+
+            for (;String.Equals(input.ToLower(), "exit") == false; input = Console.ReadLine()) 
+            {
+               
+                if (DateTime.TryParse(input, out inputDate) == true)
+                {
+                    if (inputDate > maxRange)
```

Comment:
Don't forget to check whether `inputDate` is greater than min date - `01.01.0001` ;)

Reply:
DateTime method returns false when you try parse value less then 01.01.0001. I can use manual check, but DateTime do that check automatical)

Reply:
Trust nobody ;) One day they can remove this check and values less than 01.01.0001 become valid.
For native DateTime, it's not likely, but let's assume you are using some external library. You should always validate all corner cases.